test/contracts-network/colony-expenditure.js

```diff
@@ -543,4 +548,167 @@ contract("Colony Expenditure", (accounts) => {
+    });
+
+    it("should not allow arbitration users to update expenditure fundingPotId", async () => {
+      const mask = [OFFSET];
+      const keys = [bn2bytes32(new BN(1))];
+      const value = "0x0";
+
+      await checkErrorRevert(
+        colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURES_SLOT, mask, keys, value, { from: ARBITRATOR }),
+        "colony-expenditure-bad-offset"
+      );
+    });
+
+    it("should not allow arbitration users to update expenditure domainId", async () => {
+      const mask = [OFFSET];
+      const keys = [bn2bytes32(new BN(2))];
+      const value = "0x0";
+
+      await checkErrorRevert(
+        colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURES_SLOT, mask, keys, value, { from: ARBITRATOR }),
+        "colony-expenditure-bad-offset"
+      );
+    });
+
+    it("should allow arbitration users to update expenditure finalizedTimestamp", async () => {
+      const mask = [OFFSET];
+      const keys = [bn2bytes32(new BN(3))];
+      const value = bn2bytes32(new BN(100));
+
+      await colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURES_SLOT, mask, keys, value, { from: ARBITRATOR });
+
+      const expenditure = await colony.getExpenditure(expenditureId);
+      expect(expenditure.finalizedTimestamp).to.eq.BN(100);
+    });
+
+    it("should allow arbitration users to update expenditure slot recipient", async () => {
+      const mask = [MAPPING];
+      const keys = ["0x0"];
+      const value = bn2bytes32(new BN(USER.slice(2), 16));
+
+      await colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURESLOTS_SLOT, mask, keys, value, { from: ARBITRATOR });
+
+      const expenditureSlot = await colony.getExpenditureSlot(expenditureId, 0);
+      expect(expenditureSlot.recipient).to.equal(USER);
+    });
+
+    it("should allow arbitration users to update expenditure slot claimDelay", async () => {
+      const mask = [MAPPING, OFFSET];
+      const keys = ["0x0", bn2bytes32(new BN(1))];
+      const value = bn2bytes32(new BN(100));
+
+      await colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURESLOTS_SLOT, mask, keys, value, { from: ARBITRATOR });
+
+      const expenditureSlot = await colony.getExpenditureSlot(expenditureId, 0);
+      expect(expenditureSlot.claimDelay).to.eq.BN(100);
+    });
+
+    it("should allow arbitration users to update expenditure slot payoutModifier", async () => {
+      const mask = [MAPPING, OFFSET];
+      const keys = ["0x0", bn2bytes32(new BN(2))];
+      const value = bn2bytes32(new BN(100));
+
+      await colony.setExpenditureState(1, UINT256_MAX, expenditureId, EXPENDITURESLOTS_SLOT, mask, keys, value, { from: ARBITRATOR });
+
+      const expenditureSlot = await colony.getExpenditureSlot(expenditureId, 0);
+      expect(expenditureSlot.payoutModifier).to.eq.BN(100);
+    });
+
+    it("should allow arbitration users to update expenditure slot skills", async () => {
```

Comment:
Hmm... seems tricky in this setting since to do it properly involves setting two values (the skill itself and the length of the array). Although on the other hand, it seems quirky that arbitration can edit existing skills but not add or remove any.

Reply:
Although, is there any reason why this mechanism can't be used to edit the length of the dynamic array? It's just a location in storage, after all.